### PR TITLE
Create Email Notifications

### PR DESCRIPTION
**Pull request by nateklaiber**
### Notes

Add ability to create Email Notifications.

### CODE REVIEW

**Comment by nateklaiber** [comment]
@muhallan @arthuroe - I have created our `develop` branch so I can merge in your working branches and test with all dependencies.

File: naas/models/projects.py [hunk 1]
```diff
@@ -14,3 +18,70 @@ def next(self):
...
+
+        klass_attributes = []
+
+        if request:
+            klass_attributes = request.json().get('data')
+        else:
+            Configuration.logger.error(
+                "Failure retrieving the projects {request.status_code}"
+            )
+        return cls(klass_attributes)
```
Comment (nateklaiber) [inline naas/models/projects.py:46]: @arthuroe - This returns: `NameError: name 'cls' is not defined`

File: naas/models/campaigns.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        """
+        Helper method to retrieve from the request
+        :param project_id: str
+        :param params: dict
+        :return: Campaigns
+        """
+        if params is None:
+            params = {}
+
+        request = naas.requests.Campaigns.list_by_project_id(
```
Comment (nateklaiber) [inline naas/models/campaigns.py:42]: @arthuroe - This returns `AttributeError: module 'naas.requests' has no attribute 'Campaigns'`

File: naas/models/campaigns.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        Create a new campaign
+        :param project_id: str
+        :param params: dict
+        :raises InvalidRequestError
+        :return: Campaign
+        """
+        if params is None:
+            params = {}
+
+        request = naas.requests.Campaigns.create_by_project_id(
```
Comment (nateklaiber) [inline naas/models/campaigns.py:100]: @arthuroe - This give me: `AttributeError: module 'naas.requests' has no attribute 'Campaigns'`

File: naas/models/projects.py [hunk 4]
```diff
@@ -14,3 +18,70 @@ def next(self):
...
+        request = naas.requests.Projects.create(params)
+
+        if request:
+            return Project(request.json().get('data'))
+
+        error = Error(request.json().get('data'))
+        failure_message = (
+            f"Failure creating the record {error.full_messages}")
+
+        Configuration.logger.error(failure_message)
```
Comment (nateklaiber) [inline naas/models/projects.py]: @arthuroe - This give me: `AttributeError: type object 'Configuration' has no attribute 'logger'`

**Review by nateklaiber — COMMENTED** [review]
Hi @muhallan and @arthuroe - I've tried testing the list, detail, and creates for models but get stuck with a handful of errors. I've commented in those areas. I was able to work with the `Directory` and a few other listings, but unable to create due to the errors I've run into. Let me know if there are any questions fo rme!

File: naas/models/email_notification_basic.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        """
+        request = naas.requests.EmailNotificationBasics.create_from_attributes(
+            email_address, project_id, campaign_id,
+            campaign_email_template_id, content, options
+        )
+
+        if request:
+            return EmailNotification(request.json().get('data'))
+
+        Configuration.logger.error(
```
Comment (nateklaiber) [inline naas/models/email_notification_basic.py]: @arthuroe - I get this: `NameError: name 'Configuration' is not defined`

File: naas/requests/email_notification_basics.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        if params is None:
+            params = {}
+        request_body = {
+            "email_notification_basic": params
+        }
+        rel = Client.rel_for('rels/email-notification-basic')
+        route = Client.routes().route_for(rel)
+        url = route.url_for()
+        request = Client.post(url, data=request_body)
+        return request
```
Comment (nateklaiber) [inline naas/requests/email_notification_basics.py:28]: @arthuroe - `EmailNotificationBasic.create('nate@theklaibers.com', 'project-name', 'campaign-name', 'template-name')` will give me the `Configuration` error, but when I inspect what is being sent to the API I see that it's not sending all of the attributes over. It's sending a hash that looks like this:

```
Started POST "/email-notifications/basic" for 75.179.34.110 at 2019-05-03 15:00:30 -0400
Processing by API::V1::EmailNotificationsBasicController#create as NAAS_JSON
  Parameters: {"email_notification_basic"=>"tags"}
```

As you can see, there are no other attributes in there.

**Comment by arthuroe** [comment]
@nateklaiber Let me work on resolving these errors. In regards to `post requests`, I have been getting `400` or `500`. I think it maybe an issue with the data am parsing. Is there a specific format for data sent?

**Comment by arthuroe** [comment]
@nateklaiber develop is updated with the fixes for the errors above?

**Comment by nateklaiber** [comment]
@arthuroe  - might have another circular reference issue. The latest with `develop` gives me: `ImportError: cannot import name 'EmailNotification' from 'naas.models.email_notification`

and `naas/models/email_notification.py` has `from naas.models.email_notifications import EmailNotifications`, and `naas/models/email_notifications.py` has `from naas.models.email_notification import EmailNotification`

**Pull request merged**